### PR TITLE
Handle nested folders as data entity.

### PR DESCRIPTION
**Pull request by ykobuk**

### CODE REVIEW

**Review by Vixtir — CHANGES_REQUESTED** [review]

File: odd_collector_aws/adapters/s3/adapter.py [hunk 1]
```diff
@@ -29,9 +31,14 @@ def get_data_source_oddrn(self) -> str:
         return self._oddrn_generator.get_data_source_oddrn()
 
     def get_data_entity_list(self) -> DataEntityList:
+        entities = list(self._get_entities())
```
[the remark this answers is not in the document]
Comment (ykobuk) [reply odd_collector_aws/adapters/s3/adapter.py]: Done

File: odd_collector_aws/utils/handle_nested_structure.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        return data_entity
+
+    def _parse_oddrns(self, oddrns: List[str], s3_path: str, main_oddrn: str) -> Dict[str, List[str]]:
+        """
+        Create dictionary from list of created oddrns from files,
+        where `key` is path, and `value` is a list of files
+        related to the same path.
+        """
+        result = defaultdict(list)
+        second_part_of_oddrn = [i.split(main_oddrn)[-1].replace(self.backslashes, self.delimiter) for i in oddrns]
```
[the remark this answers is not in the document]
Comment (ykobuk) [reply odd_collector_aws/utils/handle_nested_structure.py]: Done.

File: odd_collector_aws/utils/handle_nested_structure.py [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+        return result
+
+    def get_data_entity_list(self, bucket: str, s3_path: str) -> List[DataEntity]:
+        """Get combined list with DataEntity objects."""
+        self.oddrn_generator.set_oddrn_paths(
+            buckets=bucket,
+            keys=s3_path
+        )
+        oddrn_by_path = self.oddrn_generator.get_oddrn_by_path('keys')
+        self.replaced_oddrn_path = oddrn_by_path.replace(self.backslashes, self.delimiter)
```
[the remark this answers is not in the document]
Comment (ykobuk) [reply odd_collector_aws/utils/handle_nested_structure.py]: Done

File: odd_collector_aws/utils/handle_nested_structure.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (ykobuk) [reply odd_collector_aws/utils/handle_nested_structure.py]: Done.

**Pull request merged**